rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/FileDownloader.java

```diff
@@ -132,7 +161,7 @@ protected long processDownloadRequests(List<FileDownloadRequestChain> readReques
       FileDownloadRequestChain requestChain = readRequestChainList.get(futures.indexOf(future));
       long totalBytesToBeDownloaded = 0;
       for (ReadRequest request : requestChain.getReadRequests()) {
-        totalBytesToBeDownloaded += request.getBackendReadLength();
+        totalBytesToBeDownloaded += request.getBackendReadEnd() - request.getBackendReadStart();
```

Comment:
If `request.getBackendReadLength()` is not safe to use due to the cast, then can we fix that or remove that method ?

Reply:
Yes, it needed lot more changes but probably is better to do that for clarity

Reply:
Done